src/foam/u2/detail/SectionView.js

```diff
@@ -38,37 +55,71 @@ foam.CLASS({
       class: 'Boolean',
       name: 'showTitle',
       value: true
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.ArraySlot',
+      name: 'visibilitySlots_',
+      documentation: `
+        An array of slots used to control the visibility of the child
+        SectionedDetailPropertyViews of this view. Used internally to figure out
+        which children are the first and last ones visible so we can adjust the
+        padding on those elements to make sure nested
+        SectionedDetailPropertyViews don't cause unnecessary padding.
+      `,
+      factory: function() {
+        return this.ArraySlot.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'firstVisibleIndex_'
+    },
+    {
+      class: 'Int',
+      name: 'lastVisibleIndex_'
     }
   ],
   methods: [
     function initE() {
       var self = this;
-      this.addClass(this.myClass());
       self.SUPER();
-      
+      var proxySlot = self.ProxySlot.create({ delegate$: self.visibilitySlots_$ });
```

Comment:
Sweet! Ok I think we can do a tad more cleanup here. First, remove all of these new properties.
Now this part becomes:
```
var proxySlot = self.ProxySlot.create({ delegate$: self.ConstantSlot.create({value: []}) });
var firstVisibleIndex = proxySlot.map(arr => arr.indexOf(true));
var lastVisibleIndex = proxySlot.map(arr => arr.lastIndexOf(true));
```
See below for comments on other lines as a result...

Reply:
Had to do

```
var proxySlot = self.ProxySlot.create({ delegate: self.ConstantSlot.create({value: []}) });
```

instead, setting `delegate` instead of `delegate$`. Otherwise we get the following error: 

<img width="392" alt="Screen Shot 2019-07-22 at 9 10 27 AM" src="https://user-images.githubusercontent.com/4259165/61635150-ea583d00-ac60-11e9-9224-1156a4798bf8.png">


### Explanation (mostly for myself to make sure I really get it):

When you call `copyFrom` with a property slot as a key like `{ delegate$: someSlot }`, it calls https://github.com/foam-framework/foam2/blob/master/src/foam/core/Property.js#L424, and the `toSlot` part of that line creates a `PropertySlot` on whatever property you chose, in this case `delegate`. The type of the value in the `delegate` property is expected to be a `Slot`. When we `linkFrom` two slots, we say "These two slots should always have the same value. When one of them changes, update the other to match." Therefore when we link the `PropertySlot` for the `delegate` property to the `ConstantSlot` we create, we're trying to say: "The value of the `ConstantSlot` should always be the same as the value of the `PropertySlot` for the `delegate` property`". But that's not what we want because the value of the `ConstantSlot` is an array and the value of the `delegate` property should be a slot.

That also explains why it works with the other properties where the value of the property _is_ indeed a slot, such as `{ delegate$: self.visibilitySlot$ }`. In that case we can read it as "Let the `PropertySlot` of the `delegate` property to link to the `PropertySlot` of the `visibilitySlot` value, such that they always have the same value." Since the type of the value of both the `delegate` property and the `visibilitySlot` property is `Slot`, then when one changes we update the other. That's one level, the second level is that setting a property with a Slot as its value actually just `linkFrom`s the two  And the `postSet` on the `delegate` property on `ProxySlot` makes sure we're removing old listeners when the `delegate` changes.

![Slot Diagram](https://user-images.githubusercontent.com/4259165/61636995-bda62480-ac64-11e9-8f02-2bc20f5b4f13.jpg)

Reply:
Resolved by fd57cd2.

Reply:
Yes my bad that delegate$ shouldn't have been there in my code to you. Sorry about that. Nice diagram. I wanted to draw something up to help visualize it but couldn't think of a good way of doing it. Your diagram gets the point across. Nice :)